src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Microsoft.AspNetCore.Razor.LanguageServer.Test.csproj

```diff
@@ -10,12 +10,12 @@
 
   <ItemGroup>
     <None Include="xunit.runner.json" CopyToOutputDirectory="PreserveNewest" />
-    <Compile Include="..\OSSkipConditionFactAttribute.cs" LinkBase="Shared" />
   </ItemGroup>
 
   <ItemGroup>
     <ProjectReference Include="..\..\src\Microsoft.AspNetCore.Razor.LanguageServer\Microsoft.AspNetCore.Razor.LanguageServer.csproj" />
     <ProjectReference Include="..\Microsoft.AspNetCore.Razor.Test.Common.Tooling\Microsoft.AspNetCore.Razor.Test.Common.Tooling.csproj" />
+    <ProjectReference Include="..\Microsoft.CodeAnalysis.Razor.Workspaces.Test\Microsoft.CodeAnalysis.Razor.Workspaces.Test.csproj" />
```

Comment:
❗ Please don't reference test projects from other test projects. There was significant in Razor awhile back to remove dependencies between test projects and just have each reference a common project with test utilities.

Reply:
FWIW, and others might disagree with this, but when I've moved services and interfaces down to Workspaces, I haven't bothered moving the unit tests down to the workspaces unit tests. Those tests were written in the context of the language server, and our own project system etc. so testing them in isolation at the workspaces level doesn't seem to me like it adds much value. The test inputs and outputs themselves are probably valuable, and we'll definitely need to carefully consolidate things when/if we fully transition to cohosting, but the test infra is much less exciting.

Reply:
Generally, I both agree and disagree. It depends on the situation. If the tests for those Workspaces services/interfaces have language server dependences, they can be hard to break without further test infrastructure work. In that case, I'll generally leave them behind knowing that's debt that will likely need to be dealt with later. However, if they have no dependencies on the language server, I move them if I can't think of a good reason to leave them behind (other than my own dev-laziness :smile:).

That said, if you have to add a project reference from one project to another, that's a good reason to take a look and do a rethink. Layering violations like these can be hard to break later after other coupling might have occurred.

@davidwengier, your comment is more an aside and not refuting my feedback, right?

Reply:
Yeah, I was mentioning leaving the tests in place as a way to avoid moving the unit tests around, which perhaps could avoid having to add this reference. Definitely not refuting your feedback.

Reply:
I moved TagHelperServiceTestBase back to the LanguageServer.Test project, so this change is gone now.

Reply:
FWIW, I was looking at this class awhile back and I was thinking of removing it. Everything inside of it is essentially static.